docs/src/rules/func-style.md

```diff
@@ -155,6 +156,22 @@ const foo = () => {};
 
 :::
 
+### allowTypeAnnotation
+
+Examples of additional **correct** TypeScript code for this rule with the `"declaration", { "allowTypeAnnotation": true }` options:
+
+::: correct
+
+```ts
+/*eslint func-style: ["error", "declaration", { "allowTypeAnnotation": true }]*/
+
+const foo: () => void = function() {};
+
+const bar: () => void = () => {};
```

Comment:
```suggestion
const foo: Fn = function() {};

const bar: Fn = () => {};
```

I think examples where an existing type is used would be more representative for the purpose of this option.